src/application/registration/Service.ts

```diff
@@ -0,0 +1,53 @@
+import { LimitedPartnership } from "@companieshouse/api-sdk-node/dist/services/limited-partnerships";
+
+import CustomError from "../../domain/entities/CustomError";
+import RegistrationPageType from "../../presentation/controller/registration/PageType";
+import IRegistrationGateway from "../../domain/IRegistrationGateway";
+import { logger } from "../../utils";
+
+class RegistrationService {
+  registrationGateway: IRegistrationGateway;
+
+  constructor(registrationGateway: IRegistrationGateway) {
+    this.registrationGateway = registrationGateway;
+  }
+
+  async getSubmissionById(id: string): Promise<LimitedPartnership> {
+    return await this.registrationGateway.getSubmissionById(id);
+  }
+
+  async createTransactionAndFirstSubmission(
+    registrationType: RegistrationPageType,
+    data: Record<string, any>
+  ): Promise<{
+    submissionId: string;
+    transactionId: string;
+    errors?: CustomError[];
+  }> {
+    try {
+      const transactionId = await this.registrationGateway.createTransaction(
+        registrationType
+      );
+
+      const submissionId = await this.registrationGateway.createSubmission(
+        registrationType,
+        transactionId,
+        data
+      );
+
+      return { submissionId, transactionId };
+    } catch (errors: any) {
+      logger.error(
```

Comment:
i don't know if this is the correct PR but when we log we should also pass in the request Id, so that when we look at different services logs, we can match the log lines to a particular request. i.e. we can follow the request from eric->web->api->elsewhere
We could do that in a follow-up PR.

Reply:
actually it's not just the id but the CH ApiLogger interface takes the Request object. We'll have to think of a way to use it without passing the Request around.

Reply:
this is the version of CH ApiLogger without the query (.error}
if we have to log the request, we have to do it at controller level, because passing the object Request to the application layer breaks the principle of single responsability and the whole structure no longer makes sense

Reply:
If not passing it around, could it be stored in a Node JS equivalent to the Java Thread Local object and then the lower levels get it out of there, only for the purposes of logging (or even better, out log class implementation grabs it)?